lib/Parser/RegexCompileTime.cpp

```diff
@@ -2683,28 +2706,65 @@ namespace UnifiedRegex
                     Label* caseLabels = AnewArray(compiler.ctAllocator, Label, numItems);
                     // We must fixup the switch arms
                     Label switchLabel = compiler.CurrentLabel();
-                    Assert(switchSize <= Switch20Inst::MaxCases);
+
+                    Assert(switchSize <= Switch24Inst::MaxCases);
                     if (allCanSkip)
                     {
-                        if (switchSize > Switch10Inst::MaxCases)
-                            EMIT(compiler, SwitchAndConsume20Inst);
-                        else
-                            EMIT(compiler, SwitchAndConsume10Inst);
+                        if (switchSize <= Switch2Inst::MaxCases)
+                        {
+                            EMIT(compiler, SwitchAndConsume2Inst);
+                        }
+                        else if (switchSize <= Switch4Inst::MaxCases)
+                        {
+                            EMIT(compiler, SwitchAndConsume4Inst);
+                        }
+                        else if (switchSize <= Switch8Inst::MaxCases)
+                        {
+                            EMIT(compiler, SwitchAndConsume8Inst);
+                        }
+                        else if (switchSize <= Switch16Inst::MaxCases)
+                        {
+                            EMIT(compiler, SwitchAndConsume16Inst);
+                        }
+                        else if (switchSize <= Switch24Inst::MaxCases)
```

Comment:
Should this just be an `else` to match previous behavior? I know we have an assert above, but could Bad Things happen in release builds if none of these cases are hit?

Reply:
Fair enough. However, Bad Things (more like wrong behavior) might also happen if we try to serialize as a switch something with more than the max number of cases. Maybe else FailFast?

Reply:
Also there's logic to prevent entering here if we have more than the max cases required here, so in theory it can't happen. I think `else FailFast` is the right thing.